backend/src/app/rpc/mutations/files.clj

```diff
@@ -146,6 +162,35 @@
               {:id id})
   nil)
 
+(def sql:find-files
+  "select file_id 
+    from file_library_rel
+   where library_file_id=?")
+
+(defn absorb-library
+  "Find all files using a shared library, and absorb all library assets
+  into the file local libraries"
+  [conn {:keys [id] :as params}]
+  (let [library (->> (db/get-by-id conn :file id)
+                     (files/decode-row)
+                     (pmg/migrate-file))]
+    (when (:is-shared library)
+      (let [process-file
+            (fn [row]
+              (let [ts (dt/now)
+                    file (->> (db/get-by-id conn :file (:file-id row))
+                              (files/decode-row)
+                              (pmg/migrate-file))
+                    updated-data (ctf/absorb-assets (:data file) (:data library))]
+
+                (db/update! conn :file
+                            {:revn (inc (:revn file))
+                             :data (blob/encode updated-data)
+                             :modified-at ts}
+                            {:id (:id file)})))]
+
+        (dorun (->> (db/exec! conn [sql:find-files id])
+                    (map process-file)))))))
```

Comment:
The find-files ad-hoc sql query  can be solved with `db/query` helper.

The `dorun` +`map` I think can be expressed using `run!` function.

Reply:
both done